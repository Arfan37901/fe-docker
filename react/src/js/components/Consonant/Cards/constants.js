
const defaultProps = {
    id: '',
    title: '',
    label: '',
    footer: '',
    ctaLink: '',
    endTime: '',
    logoAlt: '',
    logoSrc: '',
    videoURL: '',
    badgeText: '',
    startTime: '',
    cardImage: '',
    bannerIcon: '',
    description: '',
    logoBorderColor: '',
    bannerFontColor: '',
    isBookmarked: false,
    bannerDescription: '',
    disableBookmarkIco: '',
    logoBackgroundColor: '',
    bannerBackgroundColor: '',
};

export const SquareCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const FullCardDefaultProps = {
    ...defaultProps,
    ctaLink: '',
};

export const WideCardDefaultProps = {
    ...defaultProps,
    footer: [],
    isBookmarked: false,
    disableBookmarkIco: false,
};
