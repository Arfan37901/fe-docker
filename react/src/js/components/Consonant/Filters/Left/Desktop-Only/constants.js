export const selectedItemsDefaultProps = {
    numItemsSelected: 0,
};

export const clearButtonDefaultProps = {
    clearAllFiltersText: '',
};

export const titleDefaultProps = {
    panelHeader: '',
};
