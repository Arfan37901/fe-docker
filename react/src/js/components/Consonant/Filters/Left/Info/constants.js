
/**
 * Minimal viewport width to fit desktops/laptops
 * @type {Number}
 */
export const DESKTOP_MIN_WIDTH = 1200;

export const defaultProps = {
    cardsQty: 0,
    filtersQty: 0,
    selectedFiltersQty: 0,
    windowWidth: window.innerWidth,
};
