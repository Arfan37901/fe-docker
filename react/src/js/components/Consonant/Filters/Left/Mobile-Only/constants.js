export const groupFooterDefaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
    mobileGroupTotalResultsText: '',
};

export const panelFooterDefaultProps = {
    resQty: 0,
    doneText: '',
    applyText: '',
    showTotalResults: false,
    clearAllFiltersText: '',
    showTotalResultsText: '',
    someFiltersAreSelected: false,
};

export const titleDefaultProps = {
    leftPanelMobileHeader: '',
};

export const infoDefaultProps = {
    selectedFiltersQty: 0,
    mobileFilterBtnLabel: '',
};
