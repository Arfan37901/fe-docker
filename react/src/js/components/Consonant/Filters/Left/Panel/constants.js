
/**
 * Minimal viewport width to fit desktops/laptops
 * @type {Number}
 */
export const DESKTOP_MIN_WIDTH = 1200;

export const defaultProps = {
    resQty: 0,
    filters: [],
    selectedFiltersQty: 0,
    showMobileFilters: false,
    windowWidth: window.innerWidth,
};
