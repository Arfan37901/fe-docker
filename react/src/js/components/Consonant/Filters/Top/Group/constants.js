export const defaultProps = {
    numItemsSelected: 0,
    clearFilterText: '',
};

/**
 * Minimum quantity of the top filter options to apply blur on options' wrapper
 * @type {Number}
 */
export const clipWrapperItemsCount = 9;
