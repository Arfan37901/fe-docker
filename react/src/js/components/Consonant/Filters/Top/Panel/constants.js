/**
 * Top search bar identifier
 * @type {String}
 */
export const searchId = 'top-search';


export const defaultProps = {
    resQty: 0,
    filters: [],
    showLimitedFiltersQty: false,
};
