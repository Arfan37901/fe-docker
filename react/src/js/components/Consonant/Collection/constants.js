import { DEFAULT_SHOW_ITEMS_PER_PAGE } from '../Helpers/constants';

// eslint-disable-next-line import/prefer-default-export
export const defaultProps = {
    pages: 1,
    cards: [],
    resultsPerPage: DEFAULT_SHOW_ITEMS_PER_PAGE,
};
