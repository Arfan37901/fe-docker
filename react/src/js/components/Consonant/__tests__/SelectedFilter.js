import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SelectedFilter from '../SelectedFilter';

import { DEFAULT_PROPS } from './constants/selectedFilter';

import makeSetup from './utils/settings';

const setup = makeSetup(SelectedFilter, DEFAULT_PROPS);

describe('Consonant/SelectedFilter', () => {
    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = setup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onOpen', () => {
            const {
                props: {
                    id,
                    name,
                    onClick,
                    parentId,
                },
            } = setup();

            const buttonElement = screen.getByText(name);

            fireEvent.click(buttonElement);

            expect(onClick).toBeCalled();
            expect(onClick).toHaveBeenCalledWith(parentId, id, false);
        });
    });
});
