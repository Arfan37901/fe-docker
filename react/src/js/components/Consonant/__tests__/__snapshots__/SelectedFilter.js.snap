// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/SelectedFilter Check snapshots should renders correctly 1`] = `
<button
  className="consonant-filters--selected-filter"
  data-testid="selected-filter"
  onClick={[Function]}
  type="button"
>
  filter name
</button>
`;
